### ISSUE TITLE
Wrong relationship assignment

### ISSUE THREAD

**Issue by RobinFalko** [hyperoslo/Sync#254]
Hi,

I just ran into an issue where two relationships between two types are not assigned properly. In my example I have a "House" and a "Human" with relationships:

![screen shot 2016-08-08 at 11 56 29](https://cloud.githubusercontent.com/assets/17236480/17476225/31cafafa-5d5f-11e6-8c12-9f179a1e73d8.png)

_House_
```
{
    "id":0,
    "residents":[{
        "id":0
    }]
}
```

I expect this to insert a House and a Human where the human is a resident of the house. Owners of the house should be left nil.

In my case House and Human was added correctly with the relationship between `residents` and `residenthouse` **but it also** created a relationship between `owners` and `ownhouses` with the same instances after syncing.

Is there something I need to setup to avoid this behavior?

**Comment by 3lvis** [comment]
Oh god, I'm sorry that I didn't had the chance to look at this earlier.

**Comment by 3lvis** [comment]
I'll double check.

**Comment by 3lvis** [comment]
I was able to reproduce this bug.

**Comment by 3lvis** [comment]
Fixed! To explain it a little bit better. 

Sync first will check the `houses` in the JSON. It will find one house with `id` 0. It will create the house, then it will move to `residents`.

Sync then, will go through all the residents, it will find one with `id` 0. It will make a Human entry and will connect the `residenthouse` with the `residents`. Then it will continue the loop and check for `ownhouses` and this is where the bug was located.

Sync wasn't very good at handling Entities that point to each other multiple times, so it would think that both `ownhouses` and `residenthouse` in Human are the same.

I fixed this issue by sending which relationship was supposed to be evaluated for an array of `residents`. Meaning if I'm parsing residents, then I should only connect `residenthouse` and ignore `ownhouses`.
